src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionSetupForm.php

```diff
@@ -0,0 +1,352 @@
+
+    protected const LOGIN_MODE_SINGLE_USER = 'single_user';
+    protected const LOGIN_MODE_DYNAMIC_USER = 'dynamic_user';
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return void
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $this->addLoginModeField($builder)
+            ->addCompanyBusinessUnitField($builder, [])
+            ->addCompanyBusinessUnitFieldListeners($builder)
+            ->addCompanyUserField($builder, [])
+            ->addCompanyUserFieldListeners($builder);
+    }
+
+    /**
+     * @param \Symfony\Component\OptionsResolver\OptionsResolver $resolver
+     *
+     * @return void
+     */
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => PunchoutCatalogConnectionSetupTransfer::class,
+        ]);
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return $this
+     */
+    protected function addLoginModeField(FormBuilderInterface $builder)
+    {
+        $builder->add(PunchoutCatalogConnectionSetupTransfer::LOGIN_MODE, ChoiceType::class, [
+            'label' => static::FIELD_LABEL_LOGIN_MODE,
+            'choices' => [
+                'single_user' => static::LOGIN_MODE_SINGLE_USER,
+                'dynamic_user_creation' => static::LOGIN_MODE_DYNAMIC_USER,
+            ],
+            'attr' => [
+                'class' => 'dependent-trigger',
+                'data-dependent-group' => static::DEPENDENT_GROUP_LOGIN_MODE,
+            ],
+            'constraints' => new NotBlank(),
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return $this
+     */
+    protected function addCompanyBusinessUnitField(FormBuilderInterface $builder, array $options)
+    {
+        $builder->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_BUSINESS_UNIT, SelectType::class, array_merge(
+            $this->getCompanyBusinessUnitFieldOptions(),
+            $options
+        ));
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return $this
+     */
+    protected function addCompanyUserField(FormBuilderInterface $builder, array $options)
+    {
+        $builder->add(
+            PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_USER,
+            SelectType::class,
+            array_merge($this->getCompanyUserFieldOptions(), $options)
+        );
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return $this
+     */
+    protected function addCompanyBusinessUnitFieldListeners(FormBuilderInterface $builder)
```

Comment:
So, this method does two things:
1) Creates form modification callback, and
2) Registers it to 2 events.

It looks like SRP violation. I would suggest moving callback creation to the separate `getFormModificationCallback()` method, and call it here.